external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis2/KinesisInputDStream.scala

```diff
@@ -0,0 +1,392 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.kinesis2
+
+import java.net.URI
+
+import scala.reflect.ClassTag
+
+import software.amazon.awssdk.http.Protocol
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.kinesis.model.Record
+import software.amazon.kinesis.common.InitialPositionInStream
+import software.amazon.kinesis.retrieval.KinesisClientRecord
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.storage.{BlockId, StorageLevel}
+import org.apache.spark.streaming.{Duration, StreamingContext, Time}
+import org.apache.spark.streaming.api.java.JavaStreamingContext
+import org.apache.spark.streaming.dstream.ReceiverInputDStream
+import org.apache.spark.streaming.receiver.Receiver
+import org.apache.spark.streaming.scheduler.ReceivedBlockInfo
+
+private[kinesis2] class KinesisInputDStream[T: ClassTag](_ssc: StreamingContext,
+                      val streamName: String,
+                      val endpointUrl: URI,
+                      val regionName: String,
+                      val kinesisCreds: SparkAWSCredentials,
+                      val dynamoDBCreds: Option[SparkAWSCredentials],
+                      val cloudWatchCreds: Option[SparkAWSCredentials],
+                      val cloudWatchUrl: Option[URI],
+                      val checkpointAppName: String,
+                      val checkpointInterval: Duration,
+                      val maxRecords: Option[Integer],
+                      val protocol: Option[Protocol],
+                      val initialPositionInStream: Option[InitialPositionInStream],
+                      val dynamoProxyHost: Option[String],
+                      val dynamoProxyPort: Option[Integer],
+                      val _storageLevel: StorageLevel,
```

Comment:
why have underscore?

Reply:
It was copied from KCL 1 parameter list. Anyway fixes for new version. _ is removed.